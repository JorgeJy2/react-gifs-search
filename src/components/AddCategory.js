import React,{useState} from 'react'
import PropTypes from 'prop-types';


export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('');
    
    const handleInputChange = ({target: {value}})=> {
        // console.log(value);
        setinputValue(value)
    };


    const handleSubmit = (e)=> {
       e.preventDefault();
    //    console.log('Handle submit');
       
       if(inputValue.trim().length > 2){
        //    console.log(e);

           setCategories(categories =>  [inputValue,...categories]);
           setinputValue('');
       }
    };


    return (
        <form onSubmit={handleSubmit}>
            {/* <p> Add Category</p> */}
            <input 
                placeholder="Search"
                type="text"
                value={inputValue}
                onChange={handleInputChange}
                />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};
