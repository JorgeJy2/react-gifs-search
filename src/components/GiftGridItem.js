
import React from 'react'

export const GiftGridItem = ( {url,title} ) => {
    return (
        <div className="card animate__animated animate__fadeInDown">
           <img src={url} alt={title}/>
           <p>{title}</p>
        </div>
    )
}

