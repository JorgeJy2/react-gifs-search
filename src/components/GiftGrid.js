import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifs } from './hooks/UseFetchGifs';
import { GiftGridItem } from './GiftGridItem';

export const  GiftGrid = ({category}) => {

    const {data: images,loading} = useFetchGifs(category);   
    
    return (
        <>
        <p className="category">{category}</p>
        {loading && <p id="message-loading">Loading....</p>}
        <div className="card-grid">
                {
                    images?.map((img) => (
                        <GiftGridItem 
                            key={img.id}
                            {...img}
                            />
                    ))
                }
        </div>
        </>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
};
