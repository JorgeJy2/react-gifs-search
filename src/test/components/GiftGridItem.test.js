import React from 'react';
import { shallow } from "enzyme";
import { GiftGridItem} from "../../components/GiftGridItem";

describe('GiftGridItem', () => {
    test('should view component', () => {
        const wrapper = shallow( < GiftGridItem / > );
        expect(wrapper).toMatchSnapshot();
    });
});