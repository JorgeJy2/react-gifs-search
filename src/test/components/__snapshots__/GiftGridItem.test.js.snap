// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftGridItem should view component 1`] = `
<div
  className="card animate__animated animate__fadeInDown"
>
  <img />
  <p />
</div>
`;
